### PR TITLE
CIP-24: Versioned Gas Scheduler Variables

### PR DESCRIPTION
**Pull request by ninabarbakadze**

### CODE REVIEW

File: cips/cip-24.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+| cip | 24 |
```
Comment (jcstein) [inline cips/cip-24.md:1]:
```suggestion
| cip | TBD |
```
hi @ninabarbakadze, can you please change filename to something like `cip-versioned-gas-schedule-variables.md` and then once it is approved for draft, we will assign it a number? thank you

Comment (ninabarbakadze) [reply cips/cip-24.md]: done!

File: cips/versioned_gas_scheduler_variables.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+| author | Nina Barbakadze (@ninabarbakadze) |
+| discussions-to |  |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-07-24 |
+
+## Abstract
+
+Introduce versioning for on-chain governance modifiable gas scheduler parameters, such as `blob.GasPerBlobByte` and `auth.TxSizeCostPerByte`, to only allow changes during hard fork upgrades.
```
Comment (cmwaters) [inline cips/versioned_gas_scheduler_variables.md:14]: You don't have to use "such as" when these two variables are the exhaustive list.

Gas scheduler parameters: `blob.GasPerBlobByte` and `auth.TxSizeCostPerByte` will no longer be modifiable by governance but may only change via a hard fork upgrade

File: cips/versioned_gas_scheduler_variables.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+Previously:
+
+1. `blob.GasPerBlobByte`
+2. `auth.TxSizeCostPerByte`
+
+Now:
+
+1. `appconsts.GasPerBlobByte`
+1. `appconsts.TxSizeCostPerByte`
```
Comment (cmwaters) [inline cips/versioned_gas_scheduler_variables.md:36]: Can we include what the values of these will be when implemented (in v3 potentially)

File: cips/versioned_gas_scheduler_variables.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### Rationale
+
+Versioned gas scheduler variables allow for hardcoding these values into estimators, simplifying the gas estimation process and making transaction costs more predictable without the need for pre-transaction queries.
```
Comment (cmwaters) [inline cips/versioned_gas_scheduler_variables.md:40]: Motivation and Rationale serve the same purpose in this situation (and in many others). I would advocate consolidating and having just one

File: cips/versioned_gas_scheduler_variables.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+## Backwards Compatibility
+
+Enabling this feature requires a hard fork network upgrade.
+
+## Test Cases
+
+Test cases should verify that gas scheduler variables are exclusively updated via hard forks, effectively preventing updates through governance mechanisms.
```
Comment (cmwaters) [inline cips/versioned_gas_scheduler_variables.md:48]: We'd also need to test that the gas meter uses those constants

**Review by cmwaters — APPROVED** [review]
Few minor nits, but looks good. Don't forget to create the forum post and add the link here

File: cips/cip-24.md (hunk 1 above)
Comment (jcstein) [reply cips/cip-24.md]: thank you! for context, this is just to avoid merge conflicts while this is in draft

**Review by cmwaters — APPROVED** [review]

**Comment by cmwaters** [comment]
@jcstein can we get this merged

**Comment by jcstein** [comment]
yep @cmwaters and @ninabarbakadze -- reviewing now

**Review by jcstein — APPROVED** [review]

**Comment by jcstein** [comment]
:shipit:

File: cips/versioned_gas_scheduler_variables.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+| cip | 24 |
+| - | - |
+| title | Versioned Gas Scheduler Variables |
+| description | Transition to hard fork-only updates for gas scheduler variables |
+| author | Nina Barbakadze (@ninabarbakadze) |
+| discussions-to |  |
```
Comment (jcstein) [inline cips/versioned_gas_scheduler_variables.md:6]: Needs `discussions-to: The url pointing to the official discussion thread` per CIP-1: https://cips.celestia.org/cip-1.html#discussions-to-header

**Review by jcstein — CHANGES_REQUESTED** [review]

File: cips/cip-24.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+| - | - |
+| title | Versioned Gas Scheduler Variables |
+| description | Transition to hard fork-only updates for gas scheduler variables |
+| author | Nina Barbakadze (@ninabarbakadze) |
```
Comment (jcstein) [inline cips/cip-24.md:5]:
```suggestion
| author | Nina Barbakadze ([@ninabarbakadze](https://github.com/ninabarbakadze) |
```

**Review by jcstein — APPROVED** [review]

**Pull request merged**